components/inputs/input-textarea.js

```diff
@@ -0,0 +1,261 @@
+import '../tooltip/tooltip.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { formatNumber } from '@brightspace-ui/intl/lib/number.js';
+import { FormElementMixin } from '../form/form-element-mixin.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { inputLabelStyles } from './input-label-styles.js';
+import { inputStyles } from './input-styles.js';
+import { offscreenStyles } from '../offscreen/offscreen.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+import { SkeletonMixin } from '../skeleton/skeleton-mixin.js';
+import { styleMap } from 'lit-html/directives/style-map.js';
+
+class InputTextArea extends FormElementMixin(SkeletonMixin(RtlMixin(LitElement))) {
+
+	static get properties() {
+		return {
+			/**
+			 * Indicates that the input value is invalid.
+			 */
+			ariaInvalid: { type: String, attribute: 'aria-invalid' },
+			/**
+			 * Additional information communicated in the aria-describedby on the input.
+			 */
+			description: { type: String, reflect: true },
+			/**
+			 * Disables the input.
+			 */
+			disabled: { type: Boolean, reflect: true },
+			/**
+			 * REQUIRED: Label for the input.
+			 */
+			label: { type: String },
+			/**
+			 * Hides the label visually (moves it to the input's "aria-label" attribute).
+			 */
+			labelHidden: { type: Boolean, attribute: 'label-hidden' },
+			/**
+			 * Maximum height of the input before scrolling.
+			 */
+			maxHeight: { type: String, attribute: 'max-height' },
+			/**
+			 * Imposes an upper character limit.
+			 */
+			maxlength: { type: Number },
+			/**
+			 * Minimum height of the input.
+			 */
+			minHeight: { type: String, attribute: 'min-height' },
+			/**
+			 * Imposes a lower character limit.
+			 */
+			minlength: { type: Number },
+			/**
+			 * Placeholder text.
+			 */
+			placeholder: { type: String },
+			/**
+			 * Indicates that a value is required.
+			 */
+			required: { type: Boolean, reflect: true },
+			/**
+			 * Value of the input.
+			 */
+			value: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [ super.styles, inputStyles, inputLabelStyles, offscreenStyles, css`
+			:host {
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-input-label {
+				display: inline-block;
+				vertical-align: bottom;
+			}
+			:host .d2l-input-textarea-container {
+				max-width: 100%;
+				position: relative;
+			}
```

Comment:
Is the ':host' prefix needed for all of these? I thought it was always implied.

Reply:
Thanks, these can mostly be removed.  There was one case where the `:host` was providing some necessary selectivity, but I will change it to `textarea.d2l-input` to make that clearer.